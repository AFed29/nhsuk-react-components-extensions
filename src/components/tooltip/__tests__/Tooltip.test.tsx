import React from 'react';
import { mount } from 'enzyme';
import Tooltip from '..';

describe('Tooltip', () => {
  it('Matches Snapshot', () => {
    const component = mount(<Tooltip tooltip="Tooltip!">Hover Over Me</Tooltip>);
    expect(component).toMatchSnapshot();
    component.unmount();
  });

  it('Passes through IDs', () => {
    const componentWithoutId = mount(<Tooltip tooltip="Tooltip!">Hover Over Me</Tooltip>);
    const componentWithId = mount(
      <Tooltip tooltip="Tooltip!" id="test-id">
        Hover Over Me
      </Tooltip>,
    );

    expect(componentWithoutId.find('div').prop('aria-labelledby')).toBe(undefined);
    expect(componentWithId.find('div').prop('aria-labelledby')).toBe('test-id-tooltip-text');

    expect(componentWithId.find('span').prop('id')).toBe('test-id-tooltip-text');
    expect(componentWithoutId.find('span').prop('id')).toBe(undefined);

    componentWithId.unmount();
    componentWithoutId.unmount();
  });
});
