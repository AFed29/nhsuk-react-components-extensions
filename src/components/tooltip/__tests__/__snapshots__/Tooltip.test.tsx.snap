// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip Matches Snapshot 1`] = `
<Tooltip
  textProps={Object {}}
  tooltip="Tooltip!"
>
  <div
    className="nhsuk-tooltip"
  >
    Hover Over Me
    <span
      className="nhsuk-tooltip__text"
    >
      Tooltip!
    </span>
  </div>
</Tooltip>
`;
